MapleServer2/MapleServer.cs

```diff
@@ -175,13 +176,24 @@ private static void DailyReset()
                 DatabaseManager.GameEventUserValue.Delete(userValue);
                 player.EventUserValues.Remove(userValue);
             }
+
+            // Weekly reset
+            if (DateTime.Now.DayOfWeek == DayOfWeek.Friday)
+            {
+                WeeklyReset(player);
+            }
         }
 
         DatabaseManager.RunQuery("UPDATE `characters` SET gathering_count = '[]'");
 
         DatabaseManager.ServerInfo.SetLastDailyReset(TimeInfo.CurrentDate());
     }
 
+    private static void WeeklyReset(Player player)
```

Comment:
Isn't this function missing updating all players? Not only the ones online? And it's missing the database update, no?